nion/swift/Panel.py

```diff
@@ -376,6 +395,108 @@ def context_menu_event(self, x: int, y: int, gx: int, gy: int) -> bool:
+            # Variables containing the coordinates of the rendering locations, not the full clickable control
+            close_box_left = canvas_size.width - control_width_with_margin + control_margin_x
+            close_box_right = canvas_size.width - control_margin_x
+            close_box_top = canvas_size.height // 2 - 3
+            close_box_bottom = canvas_size.height // 2 + 3
+
+            control_margin_y = canvas_size.height - (close_box_bottom - close_box_top) - 2
+            if control_margin_y > control_margin_x:
+                control_margin_y = control_margin_x
+
+            close_rect = Geometry.IntRect.from_tlhw(
+                                 close_box_top - control_margin_y,
+                                 close_box_left - control_margin_x,
+                                 control_size + 2 * control_margin_y,
+                                 control_width_with_margin) #  Rectangle containing bounds of the control
+            with drawing_context.saver():
+                drawing_context.begin_path()
+                drawing_context.move_to(close_box_left, close_box_top)
+                drawing_context.line_to(close_box_right, close_box_bottom)
+                drawing_context.move_to(close_box_left, close_box_bottom)
+                drawing_context.line_to(close_box_right, close_box_top)
+                drawing_context.line_width = 1.5
+                drawing_context.line_cap = "round"
+                drawing_context.stroke_style = self.__control_style
+                drawing_context.stroke()
+        return close_rect
+
+    def _draw_title_text(self, drawing_context: DrawingContext.DrawingContext) -> typing.Optional[Geometry.IntRect]:
+        canvas_size = self.canvas_size
+        title_rect: typing.Optional[Geometry.IntRect] = None
+        if canvas_size:
+            with drawing_context.saver():
+                drawing_context.font = self.__font
+                drawing_context.text_align = 'center'
+                drawing_context.text_baseline = 'bottom'
+                drawing_context.fill_style = '#000'
+                drawing_context.fill_text(self.title, canvas_size.width // 2, canvas_size.height - self.__text_offset)
+
+            font_string = self.__font
+            if self.__get_font_metrics:
+                font_metrics = self.__get_font_metrics(font_string, self.title)
+                title_rect = Geometry.IntRect.from_tlhw(
+                    canvas_size.height - self.__text_offset - (font_metrics.height // 2),
+                    (canvas_size.width // 2) - (font_metrics.width // 2),
+                    font_metrics.height,
+                    font_metrics.width)
+            else:
+                 # Without knowing how big the text is, need to do a clunky fallback
+                 # Close button and pencil is roughly 40 pixels wide
+                title_rect = Geometry.IntRect.from_tlhw(0, 40, canvas_size.height, canvas_size.width - 80)
+        return title_rect
+
+
```

Comment:
Should only be a single blank line here. Please consider using `flake8 --ignore=E501 nion/swift/Panel.py nion/swift/DisplayPanel.py` and addressing any issues specific to this PR. You may need to install `conda install flake8` in your environment.